src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -70,15 +72,20 @@ protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<Data
     @Override
     protected FlowSet<DataFlowAbstraction> gen(Unit u, FlowSet<DataFlowAbstraction> in) {
         FlowSet<DataFlowAbstraction> res = new ArraySparseSet<>();
+
+        filterJInstanceFields(u);
+
         if (isLeftStatement(u) || isRightStatement(u)) {
             Statement stmt = getStatementAssociatedWithUnit(u);
             u.getDefBoxes().forEach(valueBox -> {
                 if(valueBox.getValue() instanceof Local){
                     res.add(new DataFlowAbstraction((Local) valueBox.getValue(), stmt));
                 }else if(valueBox.getValue() instanceof JArrayRef){
-                    res.add(new DataFlowAbstraction((Local) getJArrayRefName((JArrayRef) valueBox.getValue()), stmt));
+                    res.add(new DataFlowAbstraction(containArrayChain(valueBox), stmt));
```

Comment:
contain Array Chain gives me an idea of ​​returning a boolean. It might be good to review that name as well.

Reply:
I changed to getJArrayRefName.